mkosi/__init__.py

```diff
@@ -1462,15 +1462,27 @@ def run_depmod(state: MkosiState) -> None:
     if state.config.bootable == ConfigFeature.disabled:
         return
 
+    outputs = (
+        "modules.dep",
+        "modules.dep.bin",
+        "modules.symbols",
+        "modules.symbols.bin",
+    )
+
+    filters = state.config.kernel_modules_include or state.config.kernel_modules_exclude
+
     for kver, _ in gen_kernel_images(state):
+        if not filters and all((state.root / "usr/lib/modules" / kver / o).exists() for o in outputs):
```

Comment:
nit: since `not filters` is the only use of `filters`, the logic would be slightly easier to follow if
```suggestion
    no_filters = not state.config.kernel_modules_include and not state.config.kernel_modules_exclude

    for kver, _ in gen_kernel_images(state):
        if no_filters and all((state.root / "usr/lib/modules" / kver / o).exists() for o in outputs):
```

Reply:
I disagree here, I much prefer what's there now.